### ISSUE TITLE
Admin Interface Editing Panes Can Be Escaped By Screen Readers

### ISSUE THREAD

**Issue by Santoz2** [ManifoldScholar/manifold#2434]
The Editing Pane in the Admin Interface shares some characteristics of modals, which pop up over content, and grays out content behind it. However, the Editing Pane differs in behavior by not trapping users' focus completely, causing great confusion for users.

## Expected Behavior
When using something like a modal, the user should be trapped inside of it, and not be able to reach the content behind it. 

Below is an example of a page with a modal that highlights expected behavior for modals. After inputting requested fields at the bottom of the page, and pressing the submit button, a modal that traps focus completely pops up. Also note how the entirety of the page is grayed out, except for the modal.
[https://www.dropbox.com/s/d55h6wzse06pz80/index.html?dl=0](https://www.dropbox.com/s/d55h6wzse06pz80/index.html?dl=0)

## Current Behavior
It is possible to break out of the Editing Pane using a screen reader's browse mode functionality. 

## Steps to Reproduce
1. Navigate to Admin Interface (via Enter Admin Mode)
2. Navigate to an editable project
3. Navigate to the Layout page
4. Navigate to and activate the edit button by Description and Images, opening an Edit Pane. Note how the banner is still visible (which is non-standard for modals)
5. Activate Browse Mode of a screen reader (if using NVDA, this may be done with the NVDA Modifier Key + Space)
6. Using up or down arrow keys, continue to navigate the pane until content from the grayed out area is read. This pane may be escaped from either the top or the bottom.

## Testing Environment
NVDA 2019.2.1
ESR Firefox 68.2.0 esr

**Comment by dananjohnson** [comment]
Thank you for your ongoing feedback, @Santoz2! I wonder if using the `dialog` role with a proper label and description would solve this?

There's some variation in our implementation of drawers that we might want to revisit. Some should probably be treated as role `dialog`, like the example you mention, and we should consider whether to elevate them over the rest of the application. We currently position most of them below the app header. In other cases, users may still want to interact with content outside the drawer, such as when editing maker and user records. We may want to move focus to the drawer when it opens in this scenario, but I’m not sure this is properly a `dialog` in which we want to disable interaction with the rest of the page.

**Comment by Santoz2** [comment]
Hi again,

In the recent meeting UW had with Terrence and Matt, we noted that there is another instance of this same issue in the reader:
![image](https://user-images.githubusercontent.com/43553368/73871103-194f8e00-4802-11ea-9fa2-a2cde9d1cef4.png)

When annotating text, attempting to add a resource opens up this pane. Like the pane in the admin interface, a screen reader can fall off the pane to the page behind it. 

One other thing we noted in the meeting was that the design between this pane and the pane in the admin interface is a bit different:

![image](https://user-images.githubusercontent.com/43553368/73871921-8fa0c000-4803-11ea-80fa-f842744cb69d.png)

In the admin interface, the pane grays out everything on the page, except for the banner. In the reader interface, the pane acts a bit more like a modal, graying out everything, including the banner. 

The behavior of the reader version of the pane is a bit closer to the expected behavior. As @dananjohnson mentioned, there may be times where a user would want to interact with content outside of the pane, but that would go against standard implementations of modals, and cause confusion. Further, implementations of this widget should be consistent.

Thanks for your time, let me know if there are any questions!

**Comment by zdavis** [comment]
Adding needs clarification label — I think we need to have some internal discussion about the inconsistency between the drawers. There are definitely cases where we don't include the overlay when a drawer is opened (for good reason), and I'm not sure I'm convinced that these need to be 100% consistent, as long as it's clear to the user what's happening. Dana's example of the user drawer as a case where we intentionally leave out the overlay is a good one. I also think the drawer serves different purpose in the reader (where it's more like a dialog box) and the backend (where it's more like a detail view), and we should keep that in mind.

In any case, I think we should remain focused on the accessibility problem here and leave the UI consistency issue for another discussion. If I understand the issue correctly, we need to prevent the screen reader from falling off the drawer if that overlay is present. As Dana suggests, fixing this may require refactoring our drawer component, which is used in a number of places, so it's probably a somewhat involved task.

**Comment by Santoz2** [comment]
Accessibility Triage Note: The main issue, that a user can escape the modal, is considered a priority 2 issue in this case. While the issue is definitely very confusing if it occurs, a user can return to the modal and complete their task, or complete the task if they never left the modal in the first place.

> In any case, I think we should remain focused on the accessibility problem here and leave the UI consistency issue for another discussion.

As for UI consistency, we consider consistency as an important principle when considering accessibility. Coming across a widget that looks the same as a previously encountered widget, a user would expect that widget to behave the same as it did before. If it does not, it can be very confusing. If a widget is intended to be different, the user should be very aware that it is different.
